design-docs/modes.md

```diff
@@ -0,0 +1,638 @@
+
+As opmodes are separate classes and there is no top-level robot class in the standard template, sharing state between opmodes is often a challenge for teams.
+
+The Java annotation-based registration approach allows for specifying a string name for the opmode (displayed on the DS); this is automatically generated from the class name if none is provided.  The annotation also allows specifying a string group name; this is used to provide display grouping of the opmodes at the DS.  Two annotations are used to separately register autonomous and teleop opmodes to filter the displayed lists on the DS for each mode.  Additionally, opmodes may be disabled through the use of an additional `@Disabled` annotation; these are hidden from the DS displayed list.
+
+# Requirements and Desirable Features
+
+For competition, matches usually consist of an autonomous period followed by a teleoperated period.  FTC and FRC currently have different amounts of operator interaction.  In FRC, the time period between auto and teleop is very short (can be <1 second some years), and so it's a requirement for FRC that both the auto and teleop modes be able to be pre-selected by the operator prior to the start of the match, and for the DS to handle automatically transitioning between these modes.  In FTC, there is currently no FMS, and so a longer delay between auto and teleop is common to allow teams to pick up controllers, but pre-selecting a teleop mode enables a more efficient transition to teleop, so having this feature benefits FTC use cases as well.
+
+It is not a requirement to support different programming languages for different operator-selectable modes (e.g. auto in Python, teleop in Java). Supporting this would require completely terminating the robot executable and starting a new robot executable (and re-initializing all the hardware), which can take a substantial and variable amount of time to complete. This is very undesirable for FRC due to the short transition from auto to teleop; although it's less problematic for FTC, it makes sharing state between modes essentially impossible, and sharing hardware configurations is also difficult.
+
+Similarly, it is not a requirement to support running the same robot project on entirely different physical robots.  That situation is better handled through entirely separate robot projects that define the unique hardware configuration.
+
+A clear enable/disable in the Driver Station that disables all robot actuators when disabled is a requirement because it is a safety-critical feature for FRC due to the size and power of FRC robots.
+
+The enabled "Init" step in the 2025 FTC SDK/DS is not a requirement due to anticipated rule changes.  However, performing initialization of modes while the robot is still disabled is a requirement for both FTC and FRC, as it's important for user code to be able to do expensive mode-specific operations (e.g. computing autonomous paths) prior to actually starting the match.
```

Comment:
Does this just mean that for FTC, you don't have to be able to move actuators or does it mean more than this?

Reply:
Yes, the main difference compared to current FTC "init" is that you won't be able to move actuators in this state--the robot is still disabled.  You can still do other things (read sensors, do computations, etc), just not drive motors.